modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -18,111 +18,142 @@ InputParameters validParams<MultiSmoothSuperellipsoidIC>()
-  params.addRequiredParam<Real>("semiaxis_b", "Mean semiaxis value in the y direction for the ellipse");
-  params.addParam<Real>("exponent", 2,"Exponent n for each superellipsoid, n=2 is a normal ellipse");
-  params.addParam<Real>("semiaxis_c", 1, "Mean semiaxis value in the z direction for the ellipse");
-  params.addParam<Real>("semiaxis_a_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the x direction for uniform, standard deviation for normal");
-  params.addParam<Real>("semiaxis_b_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the y direction for uniform, standard deviation for normal");
-  params.addParam<Real>("semiaxis_c_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the z direction for uniform, standard deviation for normal");
+  params.addRequiredParam<std::vector<Real> >("semiaxis_a", "Vector of mean semiaxis values in the x direction for the ellipse");
+  params.addRequiredParam<std::vector<Real> >("semiaxis_b", "Vector of mean semiaxis values in the y direction for the ellipse");
+  params.addRequiredParam<std::vector<Real> >("semiaxis_c", "Vector of mean semiaxis values in the z direction for the ellipse, must be set to 1 if 2D.");
+  params.addParam<std::vector<Real> >("exponent", std::vector<Real>(),"Vector of exponents for each superellipsoid, n=2 is a normal ellipse");
+  params.addParam<std::vector<Real> >("semiaxis_a_variation", std::vector<Real>(), "Vector of plus or minus fractions of random variation in the bubble semiaxis in the x direction for uniform, standard deviation for normal");
+  params.addParam<std::vector<Real> >("semiaxis_b_variation", std::vector<Real>(), "Vector of plus or minus fractions of random variation in the bubble semiaxis in the y direction for uniform, standard deviation for normal");
+  params.addParam<std::vector<Real> >("semiaxis_c_variation", std::vector<Real>(), "Vector of plus or minus fractions of random variation in the bubble semiaxis in the z direction for uniform, standard deviation for normal. Must be set to 0 if 2D.");
   MooseEnum rand_options("uniform normal none","none");
   params.addParam<MooseEnum>("semiaxis_variation_type", rand_options, "Type of distribution that random superellipsoid semiaxes will follow");
+  MooseEnum extreme_options("False True","False");
+  params.addParam<MooseEnum>("check_extremes", extreme_options, "Check all Superellipsoid extremes (center +- each semiaxis) for overlap, must have prevent_overlap set to True.");
+  MooseEnum overlap_options("False True", "False");
+  params.addParam<MooseEnum>("prevent_overlap", overlap_options, "Check all Superellipsoid centers for overlap with other Superellipsoids.");
   return params;
 }
 
 MultiSmoothSuperellipsoidIC::MultiSmoothSuperellipsoidIC(const InputParameters & parameters) :
     SmoothSuperellipsoidBaseIC(parameters),
-    _numbub(getParam<unsigned int>("numbub")),
```

Comment:
You deleted the initialization for `_numbub`.  All class variables need to be initialized in the constructor.